agent/src/main/scala/za/co/absa/atum/agent/model/Measure.scala

```diff
@@ -50,122 +49,124 @@ object AtumMeasure {
   object DistinctRecordCount {
-    private [agent] val measureName: String = "distinctCount"
-    def apply(controlCol: String): DistinctRecordCount = DistinctRecordCount(measureName, controlCol)
+    private[agent] val measureName: String = "distinctCount"
+    def apply(controlCols: Seq[String]): DistinctRecordCount = DistinctRecordCount(measureName, controlCols)
   }
 
   case class SumOfValuesOfColumn private (measureName: String, controlCol: String) extends AtumMeasure {
+    private val columnAggFn: Column => Column = column => sum(column)
 
     override def function: MeasurementFunction = (ds: DataFrame) => {
-      val aggCol = sum(col(valueColumnName))
-      val resultValue = aggregateColumn(ds, controlCol, aggCol)
-      ResultOfMeasurement(resultValue, resultValueType)
+      val dataType = ds.select(controlCol).schema.fields(0).dataType
+      val resultValue = ds.select(columnAggFn(castForAggregation(dataType, col(controlCol)))).collect()
+      ResultOfMeasurement(handleAggregationResult(dataType, resultValue(0)(0)), resultValueType)
     }
 
+    override def controlColumns: Seq[String] = Seq(controlCol)
     override val resultValueType: ResultValueType.ResultValueType = ResultValueType.BigDecimal
   }
   object SumOfValuesOfColumn {
-    private [agent] val measureName: String = "aggregatedTotal"
+    private[agent] val measureName: String = "aggregatedTotal"
     def apply(controlCol: String): SumOfValuesOfColumn = SumOfValuesOfColumn(measureName, controlCol)
   }
 
   case class AbsSumOfValuesOfColumn private (measureName: String, controlCol: String) extends AtumMeasure {
+    private val columnAggFn: Column => Column = column => sum(abs(column))
 
     override def function: MeasurementFunction = (ds: DataFrame) => {
-      val aggCol = sum(abs(col(valueColumnName)))
-      val resultValue = aggregateColumn(ds, controlCol, aggCol)
-      ResultOfMeasurement(resultValue, resultValueType)
+      val dataType = ds.select(controlCol).schema.fields(0).dataType
+      val resultValue = ds.select(columnAggFn(castForAggregation(dataType, col(controlCol)))).collect()
+      ResultOfMeasurement(handleAggregationResult(dataType, resultValue(0)(0)), resultValueType)
     }
 
-    override val resultValueType: ResultValueType.ResultValueType = ResultValueType.Double
+    override def controlColumns: Seq[String] = Seq(controlCol)
+    override val resultValueType: ResultValueType.ResultValueType = ResultValueType.BigDecimal
   }
-  object AbsSumOfValuesOfColumn  {
-    private [agent] val measureName: String = "absAggregatedTotal"
+  object AbsSumOfValuesOfColumn {
+    private[agent] val measureName: String = "absAggregatedTotal"
     def apply(controlCol: String): AbsSumOfValuesOfColumn = AbsSumOfValuesOfColumn(measureName, controlCol)
   }
 
   case class SumOfHashesOfColumn private (measureName: String, controlCol: String) extends AtumMeasure {
-
+    private val columnExpression: Column = sum(crc32(col(controlCol).cast("String")))
```

Comment:
Could this function also support a list of columns, concatenated.
Extra bonus is `*` for all columns. 😉